js/compiler/compiler.go

```diff
@@ -194,16 +243,57 @@ func newBabel() (*babel, error) {
 	return result, err
 }
 
-// Transform the given code into ES5, while synchronizing to ensure only a single
+func increaseMappingsByOne(sourceMap []byte) ([]byte, error) {
+	var err error
+	m := make(map[string]interface{})
+	if err = json.Unmarshal(sourceMap, &m); err != nil {
+		return nil, err
+	}
+	mappings, ok := m["mappings"]
+	if !ok {
+		// no mappings, no idea what this will do, but just return it as technically we can have sourcemap with sections
+		// TODO implement incrementing of `offset` in the sections?
+		// TODO (kind of alternatively) drop the newline in the "commonjs" wrapping and have only the first line wrong
+		// except everything being off by one
+		return sourceMap, nil
+	}
+	if str, ok := mappings.(string); ok {
+		// ';' is the separator between lines so just adding 1 will make all mappings be for the line after which they were
+		// originally
+		m["mappings"] = ";" + str
+	} else {
+		// we have mappings but it's not a string - this is some kind of error
+		return nil, errors.New(`missing "mappings" in sourcemap`)
```

Comment:
I don't completely understand what's going on here and _why_ this function is necessary, but from what I can see, if we hit this error, k6 will actually abort the script, right? `couldntLoadSourceMap` won't be set and sourcemaps ignored, we'll just exit with an error?

Reply:
yes if we get somehow a sourcemap that does have "mappings" - arguably the thing that a sourcemap is for (most other fields aren't really necessary always) and it happens to not be a string we will error out. I guess we can disable the sourcemap in that case as well and log it :shrug:

Reply:
:+1: a warning seems like the better approach than to completely fail

Reply:
44c2849   should fix it and I will try to write a test for this